File: apps/api/app/schemas/integrations.py
from typing import Literal

from pydantic import BaseModel


class WorkersAIStatusResponse(BaseModel):
    provider: Literal["cloudflare_workers_ai"] = "cloudflare_workers_ai"
    enabled: bool
    configured: bool
    model: str
    external_data_policy: str = "仅允许显式合成探针；不会自动发送导入数据或个人信息。"


class WorkersAIUsage(BaseModel):
    prompt_tokens: int | None = None
    completion_tokens: int | None = None
    total_tokens: int | None = None


class WorkersAIProbeResponse(BaseModel):
    provider: Literal["cloudflare_workers_ai"] = "cloudflare_workers_ai"
    model: str
    token_status: Literal["active"]
    reachable: Literal[True] = True
    sentinel_matched: bool
    usage: WorkersAIUsage
    message: str

File: apps/web/src/types/integrations.ts
export interface WorkersAIStatus {
  provider: "cloudflare_workers_ai";
  enabled: boolean;
  configured: boolean;
  model: string;
  external_data_policy: string;
}

export interface WorkersAIProbeResult {
  provider: "cloudflare_workers_ai";
  model: string;
  token_status: "active";
  reachable: true;
  sentinel_matched: boolean;
  usage: {
    prompt_tokens: number | null;
    completion_tokens: number | null;
    total_tokens: number | null;
  };
  message: string;
}
